vmdb/db/migrate/20150414094834_create_container_projects.rb

```diff
@@ -0,0 +1,18 @@
+class CreateContainerProjects < ActiveRecord::Migration
+  def up
+    create_table :container_projects do |t|
```

Comment:
> How these are different from namespaces?
> 
> I think that even when introducing OpenShift we shouldn't fall into the mistake to represent 1:1 its entities. For example projects seems to be essentially namespaces.
> 
> I think we should prioritize the namespace collection to begin with and then use projects to augment the namespace information (if there's any valuable information to add).

namespace collection effort is being developed in parallel.

Here's my take on namespace vs project - currently they both should be collected and persisted.
Projects and namespaces are indeed quite similar to each other. they also share (imo - unfortunately, they are different object types after all) the same uid.
Currently, projects have additional attributes and annotations that namespaces don't have. While both k8s and ose are not stable, it's hard to predict how far those objects will grow from each other.
Also, each project is currently backed by a namespace, but it's not guaranteed afaik that each namespace has a parallel project, even in an openshift env.
Please also keep in mind, that manageIQ UI has static menus for all entities that doesn't change based on each provider selection, and the entities menu is displayed for both k8s and ose providers together, as they are both considered a container based provider.
Hence, while we can represent/navigate from each provider to its relevant entities (from k8s to ns, from ose to project), having only projects in the menu is misleading. (especially if a user has only k8s environments and doesn't even know what a project is)
Once k8s and openshift stabilize behavior and api, we can revisit the design.

Adding @derekwaynecarr for additional insights.